### PR TITLE
fix!: Add dedicated error for stale commits and proposals [WPB-6436]

### PR DESCRIPTION
**Pull request by OtaK**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by augustocdias — DISMISSED** [review]

**Review by typfel — APPROVED** [review]

**Comment by OtaK** [comment]
Depends on #476

**Review by beltram — APPROVED** [review]

File: crypto/src/mls/conversation/decrypt.rs [hunk 1]
```diff
@@ -295,10 +300,16 @@ impl MlsConversation {
...
                     if is_duplicate {
                         CryptoError::DuplicateMessage
-                    } else if msg_epoch == self.group.epoch().as_u64() + 1 {
+                    } else if msg_epoch == group_epoch + 1 {
                         // limit to next epoch otherwise if we were buffering a commit for epoch + 2
                         // we would fail when trying to decrypt it in [MlsCentral::commit_accepted]
                         CryptoError::BufferedFutureMessage
+                    } else if msg_epoch < group_epoch {
+                        match content_type {
+                            ContentType::Application => CryptoError::WrongEpoch,
```
Comment (typfel) [inline crypto/src/mls/conversation/decrypt.rs:309]: Shouldn't this be `MessageEpochTooOld`?

Comment (OtaK) [reply crypto/src/mls/conversation/decrypt.rs:309]: @typfel Could be. But we were not returning MessageEpochTooOld before in this very specific case (we were returning WrongEpoch) so in the spirit of keeping things consistent, I kept it as-is

Comment (typfel) [reply crypto/src/mls/conversation/decrypt.rs:309]: Ok, thought we were changing the behavior. The it's fine.

**Pull request merged**